src/json.rs

```diff
@@ -45,7 +46,7 @@ impl JsonValue {
     pub fn u128(&self, key: &str) -> Result<u128, ()> {
         match self {
             JsonValue::Object(o) => match o.get(key).ok_or(())? {
-                JsonValue::Number(n) => Ok(*n as u128),
+                JsonValue::String(n) => Ok(n.parse::<u128>().expect_utf8(b"ERR_FAILED_PARSE_U128")),
```

Comment:
I don't think we should panic in this case. It's better to return the error instead.

Reply:
If we change Err result, it will be logically true to change it for all other `json parser` errors. And also for that, we should have more complicated Error kinds for that. I think it's really redundant for that simple case.

Reply:
We currently don't panic in any other function in `json` module. So if we don't introduce any new Error types in this PR, let's just have `n.parse::<u128>().map_err(|_| ())`. I think it makes no sense just to panic for `u128` and don't panic for all other types.

Reply:
I disagree. We can check in tests that specific case. This is an exception to the rule.
Again I think that's redundant.

Reply:
Per [the Rust book](https://doc.rust-lang.org/book/ch09-03-to-panic-or-not-to-panic.html), returning a `Result` instead of panicking should be the default choice because it give more control at the call-site on how to handle the error. The only time you `panic!` is when there is no possible way to recover.

In this particular case we are talking about a general JSON parsing function, so making the choice that it is impossible to recover from failed parsing seems wrong to me. I know in the current way it is used this is the case, but what if in the future we want to use this function in another context where it makes to have a fallback value in the case the parsing fails?

tl;dr I agree with @sept-en that we should use `Result` error variant instead of `panic` here.

Reply:
As a side note: why are we removing the `Number` match branch? Couldn't this function return a value if there is a `Number` or if there is a `String` which can be parsed into a number?

Reply:
Number is an unreachable case.

Reply:
As I mentioned we should introduce custorm errors for all parser cases if we want return Result with specific error.

Reply:
Solved after merging #131 in 0dae64c